connectors/sources/sharepoint_online.py

```diff
@@ -983,6 +1005,43 @@ def _dls_enabled(self):
 
         return self._features.document_level_security_enabled()
 
+    def access_control_query(self, access_control):
+        return {"query": {"template": {"params": {"access_control": access_control}}}}
```

Comment:
Should this be a private method? e.g. `_access_control_query`, as this is called from a private method. Does it need to be present in the `BaseDataSource` class?

Reply:
No as we'll need that later on for the custom role provider to work. Every data source supporting DLS need to implement this.